File: 409-longest-palindrome/longest-palindrome.c
#include <string.h>

int longestPalindrome(char * s) {
    int charCount[128] = {0};  // Assuming ASCII characters
    int length = strlen(s);
    
    for (int i = 0; i < length; i++) {
        charCount[s[i]]++;
    }
    
    int palindromeLength = 0;
    int hasOdd = 0;
    
    for (int i = 0; i < 128; i++) {
        palindromeLength += charCount[i] / 2 * 2;
        if (charCount[i] % 2 == 1) {
            hasOdd = 1;
        }
    }
    
    return palindromeLength + hasOdd;
}

File: 409-longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] charCount = new int[128];  // Assuming ASCII characters
        
        for (char c : s.toCharArray()) {
            charCount[c]++;
        }
        
        int length = 0;
        boolean hasOdd = false;
        
        for (int count : charCount) {
            length += count / 2 * 2;
            if (count % 2 == 1) {
                hasOdd = true;
            }
        }
        
        return length + (hasOdd ? 1 : 0);
    }
}